website/blog/2025/08/08-13-keeping-track-of-your-resources-with-inventory.md

```diff
@@ -0,0 +1,539 @@
+
+Example output:
+
+``` shell
+shoot--foo--worker-a-z1-996c4-rbwtz is running
+shoot--bar--worker-hln3f-z2-6d5fb-mxv6d is running
+shoot--baz--worker-qlbgk-z1-b67b7-t56wt is terminated
+```
+
+The `inventory model query` command also understands relationships between
+resources, which allows us to generate more complex reports by loading resources
+as well as any related to them resources.
+
+This command for example can be used to generate a report of the collected
+Gardener `Project` resources, but also loads the `Shoot` and `Member`
+relationships, allowing us to report on related data.
+
+``` shell
+inventory model query \
+    --model g:model:project \
+    --relation Shoots \
+    --relation Members \
+    --template-file gardener-projects-report.tmpl
+```
+
+Additional examples on how to query the data in Inventory by using SQL queries
+or the CLI can be found in the following documents.
+
+- [Database Queries](https://github.com/gardener/inventory/blob/main/docs/db-queries.md)
+- [Querying Models](https://github.com/gardener/inventory/blob/main/docs/ops-guide.md#querying-models)
+
+The provided
+[docker-compose.yaml](https://github.com/gardener/inventory/blob/main/docker-compose.yaml)
+also comes with a [pgadmin](https://www.pgadmin.org/) service, which you can use
+to query and inspect the schema, which is handy when you need to familiarize
+yourself with the existing schema.
+
+![Inventory AWS Schema](./images/inventory/aws-schema.png)
+
+In order to get better insights of the data collected by Inventory you can
+connect your Inventory database with [Metabase](https://www.metabase.com/),
+[Grafana](https://grafana.com/) or [Apache Superset](https://superset.apache.org/).
+
+This would make it possible to create nice reports and provide various
+dashboards for your users, if needed.
+
+Additionally, you may want to connect the Inventory database with your [ITSM
+system](https://en.wikipedia.org/wiki/IT_service_management), if needed. That
+way Inventory can be used to enable various IT processes, if there is such a
+need.
+
+Besides the data collected from various datasources Inventory exposes metrics
+about it's inner workings, which can be scraped for example by
+[Prometheus](https://prometheus.io/) and alerted on as well.
+
+The following screenshot shows a summary of some of the data collected from a
+Gardener landscape.
+
+![Gardener Resources Summary](./images/inventory/gardener-stats.png)
+
+This screenshot here displays a map of resources identified by Inventory as
+being orphan.
+
+![AWS leaked resources map](./images/inventory/aws-leaked-map.png)
+
+And here we can see some of the `worker` metrics showing us information about
+the AWS-related tasks.
+
+![Worker Metrics](./images/inventory/worker-metrics.png)
+
+Hooking up Inventory with your alerting system allows for early notification of
+orphan resources as soon as they are being discovered. Here we can see Inventory
+posting an alert about orphan EC2 instances in Slack.
```

Comment:
Hooking up Inventory with your alerting system allows for early notification of
orphaned resources as soon as they are discovered. Here, we can see Inventory
posting an alert about orphaned EC2 instances in Slack.